File: c++/web_tutorial_mastercopy/tut_us_map_overlay.cpp
#include <string>

using namespace std;

#include "Bridges.h"
#include "DataSource.h"
#include "data_src/USState.h"
#include "data_src/USCounty.h"
#include "USMap.h"

using namespace bridges;

// this program illustrates how to access the data of the US state county
// maps and apply attributes to them
int main(int argc, char **argv) {

	// create bridges object
#if TESTING
	// command line args provide credentials and server to test on
	Bridges bridges (atoi(argv[1]), argv[2], argv[3]);
	if (argc > 4)
		bridges.setServer(argv[4]);
#else
	Bridges bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");
#endif

	// set title
	bridges.setTitle("Overlaying Data Structure on a US Map");
	bridges.setDescription("All US states, No Counties");
	
	//Some BRIDGES visualization with location
	SLelement<string>  *el0 = new SLelement<string> ("Charlotte", "Go Niners!");
	el0->setLocation(-80.8431, 35.2271); //35.2271N, 80.8431W
	el0->setSize(.1);
	bridges.setDataStructure(el0);

	//Getting map
	DataSource ds(&bridges);
	vector<bridges::dataset::USState> map_data = ds.getUSMapCountyData (); 

	//Styling Mecklenburg County, NC
	for (auto& s : map_data) {
	  if (s.getStateName() == "North Carolina") {
	    auto& mecklenburg = s.accessCounties()["37119"]; //access by fips code
	    mecklenburg.setFillColor(Color(0,255,0));
	  }	    
	}
	//Setting Map overlay
	USMap us_maps(map_data);
	bridges.setMap (us_maps);

	//Visualize!
	bridges.visualize();

	delete el0;
	
	return 0;
}

File: python/web_tutorial_mastercopy/tut_us_map_overlay.py
from bridges.bridges import *
from bridges.data_src_dependent import *
from bridges.dl_element import *
from bridges.us_map import *

def main():
    # create the Bridges object, set credentials
#if TESTING
    # command line args provide credentials and server to test on
    args = sys.argv[1:]
    bridges = Bridges(int(args[0]), args[1], args[2])
    if len(args) > 3:
        bridges.connector.set_server(args[3])
#else
    bridges = Bridges(YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY")
#endif

    # set title, description
    bridges.set_title("Test of US Map with State Boundaries")

    #pass no parameters for the whole map
    state_info = get_map(['iowa', 'california', 'north carolina'])

    my_map = USMap(state_info)
    bridges.set_map(my_map)

    #placeholder for assigning a viz type to BRIDGES
    el0 = DLelement()
    el0.set_location(-80.8431, 35.2271) # Charlotte, NC: 35.2271N, 80.8431W

    #visualizes iowa and california with counties
    bridges.set_data_structure(el0)
    bridges.visualize()




if __name__ == '__main__':
    main()
